src/virtual_ptdf_calculations.jl

```diff
@@ -1,4 +1,74 @@
-# TODO: add descriptions to functions
+struct RowCache
+    temp_cache::Dict{Int, Array{Float64}}
+    persistent_cache_keys::Set{Int}
+    max_cache_size::Int
+    max_num_keys::Int
+end
+
+function RowCache(max_cache_size::Int, persistent_rows::Set{Int}, row_size)
+    if (length(persistent_rows) + 1) * row_size > ROW_PERSISTENT_CACHE_WARN
+        @warn "The minimum cache size for the persisted row is over 1 GiB. This specification will cause large memory usage"
+    elseif (length(persistent_rows) + 1) * row_size > max_cache_size
+        @info "The minimum cache size for the persisted row is greater than the maximum cache size. Ignoring max_cache_size = $(max_cache_size) MiB value"
+    else
+        @debug "required cache for persisted values = $((length(persistent_rows) + 1)*row_size). Max cache specification = $(max_cache_size)"
+    end
+    max_num_keys = max(length(persistent_rows) + 1, floor(Int, max_cache_size / row_size))
```

Comment:
Related to above comments, max_num_keys may be in conflict with max_cache_size. Maybe you don't need max_cache_size as a struct field.